homework/questions.txt

```diff
@@ -3,72 +3,134 @@ we went through in class.
+  
 
   i, What is a file or folder is not empty ? What flag would you use to remove it anyway?
+  rm -rf "name of the folder"
+  
 
   j, How do you move a folder/file ? What arguments does the command take?
+  mv "name of the file" "name of the folder"/"name of the file"
+  
 
   k, How can you find out more about a command and see the available options, syntax etc.
     (How/Where do you see the manual/help for a command)
+    man "command"
+    
 
   l, With what command do you rename a folder/file in the command line?
+  mv "name of the file/folder" "new name of the file/folder".
+  
 
   m, How do you copy a folder ?
+  cp -r "name of the folder" "name of the folder copy"
 
   n, What command do you use to open a folder/file ?
+  cat or less
 
 2. Creating a new project
   a, How do you create a new folder (what command) ?
+  mkdir "name of the folder"
+  
 
   b, What command do you use to create/initiate a local repository (from the command line)?
      /One can also create a new repository on Github and then just clone it to their machine./
+     git init
+     
 
   c, What command do you use to create a new file ?
+  touch "name of the file"
+  
 
   d, If you create a file and and add text to it, these changes are now on your working directory.
      How would you save these to the local repository (describe the steps with commands)
+     fork
+     git clone
+     git add "name of the file"
+     git status
+     open file on VSC to edit it and add some text then save the changes
+     git commit -m
+     git push origin master
+     
 
     - from working directory to the staging area
+    git add "name of the folder/file"
+    
 
     - from the staging area to local repository
+    git push -u origin branch_name
+    
 
   e, What does git status do ?
+  It displays the state of the working directory and the staging area. It lets you see which changes have been staged, which haven't, and which files aren't being tracked by Git.
+  
 
   f, How would you check your commit history?
+  git log
+  
 
   g, To share your project(folder), you will need to create a remote repository.
      This can be done either on Github or in the command line using "gh repo create" command.
      To connect your local repository to the remote one (if you were to create it on github)
      you need to add remote to it. Briefly describe, what would be the commands to do this ?
 
     - How do you check whether you already have some github/remote repository connected to your local one?
+    git remote -v
+    
 
     - What command do you use to add the remote
+    git remote add origin "link of the remote repo"
+    
 
   h, When local and remote repository are connected, one can push the changes/files so they are visible
      on github. What would be the structrure of the command ? Do you need to create a pull request?
+     git push -u origin master. Then you need to do a pull request on github.
```

Comment:
you would use the flag `-u` only once, when setting the origin master as "upstream", Then you can just do git push (and it will push to the origin master you set as "upstream"). Or just do `git push origin master`(or any other branch you want to be pushing to) everytime :)